### PR TITLE
Enable fallback synthetic source for token_count

### PR DESCRIPTION
**Pull request by lkts**
Enable fallback synthetic source for token_count and add tests.

### CODE REVIEW

File: modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/TokenCountFieldMapper.java [hunk 1]
```diff
@@ -124,6 +125,11 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             }
             return new DocValueFetcher(docValueFormat(format, null), context.getForField(this, FielddataOperation.SEARCH));
         }
+
+        @Override
+        public BlockLoader blockLoader(BlockLoaderContext blContext) {
+            return null;
```
Comment (lkts) [inline modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/TokenCountFieldMapper.java:131]: `token_count` is not supported in ESQL but `TokenCountFieldType` inherits `blockLoader` implementation from `NumberFieldType` which leads to fun test failures. IMO it is more logical to do this until it is actually supported.

Comment (lkts) [reply modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/TokenCountFieldMapper.java:131]: Okay, that was not a good idea it seems.

Comment (lkts) [reply modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/TokenCountFieldMapper.java:131]: Tests are okay, i was just confused.

**Review by martijnvg — APPROVED** [review]
LGTM - Typically token_count field type is configured as a multi-field, so it isn't used as part of synthetic source. This change makes it possible to synthesize source in the case a token_count field is configured a main field.

**Comment by lkts** [comment]
@elasticmachine update branch

**Pull request merged**